### PR TITLE
Added Content collection UI Widget.

### PR DESCRIPTION
**Pull request by sonnykt**
Requires https://github.com/dpc-sdp/tide_api/pull/69

image
image
image
image

On-behalf-of: @salsadigitalauorg (joshua.fernandes@salsadigital.com.au, sonny@salsadigital.com.au)

### CODE REVIEW

File: modules/tide_content_collection/tide_content_collection.install [hunk 1]
```diff
@@ -15,4 +18,65 @@ function tide_content_collection_install($is_syncing) {
...
+  $paragraph_type = ParagraphsType::load('content_collection');
+  if ($paragraph_type) {
+    $paragraph_type->set('label', 'Content Collection (Raw JSON)');
+    $paragraph_type->save();
+  }
+
+  $configs = [
+    'paragraphs.paragraphs_type.content_collection_enhanced' => 'paragraphs_type',
+    'field.field.paragraph.content_collection_enhanced.field_content_collection_config' => 'field_config',
+    'core.entity_view_display.paragraph.content_collection_enhanced.default' => 'entity_view_display',
```
Comment (sonnykt) [inline modules/tide_content_collection/tide_content_collection.install:44]: @joshua-salsadigital the update failure above https://github.com/dpc-sdp/tide_api/pull/78/files#r876563005 may be caused by this.

**Comment by sonnykt** [comment]
@MdNadimHossain This PR adds the new widget to the original work of content collection (https://github.com/dpc-sdp/tide_api/pull/69). Please merge when you are ready to release.

File: modules/tide_content_collection/tide_content_collection.module [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function tide_content_collection_paragraph_create_access(AccountInterface $account = NULL, array $context = [], $entity_bundle = NULL) {
+  $type = $entity_bundle;
+  if ($type === 'content_collection') {
+    return AccessResult::allowedIfHasPermission($account, 'access content_collection paragraph');
+  }
+  return AccessResult::neutral();
+}
```
Comment (MdNadimHossain) [inline modules/tide_content_collection/tide_content_collection.module:34]: Hi @joshua-salsadigital 
Just trying to understand the permission scenario here. So looks like it creates a custom permission for the old content_collection paragraph, isn't it ? if the permission is set for specific user role then they can create, update or delete this component. 
So my question is - why only for old content collection and not for the new one and no permission for content_collection_enhanced ?
- Also I have this test branch in [reference site](https://nginx-php.pr-83.content-reference-sdp-vic-gov-au.sdp2.sdp.vic.gov.au/) and the permission is set only administrator and site_admin role but the editor still can add the old component. Unless I am understanding or setting up the permission wrongly. So just wanted to clarify these things.

File: modules/tide_content_collection/config/install/field.field.paragraph.content_collection_enhanced.field_content_collection_config.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.paragraph.field_content_collection_config
+    - paragraphs.paragraphs_type.content_collection_enhanced
+  module:
+    - tide_content_collection
+id: paragraph.content_collection_enhanced.field_content_collection_config
```
[the remark this answers is not in the document]
Comment (MdNadimHossain) [reply modules/tide_content_collection/config/install/field.field.paragraph.content_collection_enhanced.field_content_collection_config.yml:9]: @joshua-salsadigital any updates about this issue ?

**Review by MdNadimHossain — COMMENTED** [review]
Hey @joshua-salsadigital just few queries about the permission changes and also about the longer id for the paragrpah field.

File: modules/tide_content_collection/tide_content_collection.module (hunk 2 above)
Comment (sonnykt) [reply modules/tide_content_collection/tide_content_collection.module:34]: @MdNadimHossain that permission was created after a discussion with @anthony-malkoun. The Raw JSON component (the original component) is too powerful and we don't encourage editors to have access to that component.

I doubt if your reference site has the changes from this PR. The new component doesn't exist on your site: https://nginx-php.pr-83.content-reference-sdp-vic-gov-au.sdp2.sdp.vic.gov.au/admin/structure/types/manage/landing_page/fields/node.landing_page.field_landing_page_component

Comment (sonnykt) [reply modules/tide_content_collection/tide_content_collection.module:34]: @MdNadimHossain my mistake, it seems that you rename the old component.

File: modules/tide_content_collection/tide_content_collection.module [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * Implements hook_ENTITY_TYPE_create_access().
+ */
+function tide_content_collection_paragraph_create_access(AccountInterface $account = NULL, array $context = [], $entity_bundle = NULL) {
+  $type = $entity_bundle;
+  if ($type === 'content_collection') {
+    return AccessResult::allowedIfHasPermission($account, 'access content_collection paragraph');
```
Comment (sonnykt) [inline modules/tide_content_collection/tide_content_collection.module:31]: @joshua-salsadigital this line must be `deny if not have permission`. `allowedIfHasPermission` will return an AccessResultNeutral if the condition is not met.

Comment (joshua-salsadigital) [reply modules/tide_content_collection/tide_content_collection.module]: Updated the logic to deny if do not have permission

**Review by MdNadimHossain — APPROVED** [review]

**Pull request merged**